devicetypes/stelpro/stelpro-maestro-thermostat.src/stelpro-maestro-thermostat.groovy

```diff
@@ -0,0 +1,617 @@
+	my_writeAttribute(0x201, 0x4001, 0x29, tempToSendInString, ["mfgCode": "0x1185"])
+}
+
+def increaseHeatSetpoint() {
+	float currentSetpoint = device.currentValue("heatingSetpoint")
+	def locationScale = getTemperatureScale()
+	float maxSetpoint
+	float step
+
+	if (locationScale == "C") {
+		maxSetpoint = 30;
+		step = 0.5
+	}
+	else {
+		maxSetpoint = 86
+		step = 1
+	}
+
+	if (currentSetpoint < maxSetpoint) {
+		currentSetpoint = currentSetpoint + step
+		quickSetHeat(currentSetpoint)
+	}
+}
+
+def decreaseHeatSetpoint() {
+	float currentSetpoint = device.currentValue("heatingSetpoint")
+	def locationScale = getTemperatureScale()
+	float minSetpoint
+	float step
+
+	if (locationScale == "C") {
+		minSetpoint = 5;
+		step = 0.5
+	}
+	else {
+		minSetpoint = 41
+		step = 1
+	}
+
+	if (currentSetpoint > minSetpoint) {
+		currentSetpoint = currentSetpoint - step
+		quickSetHeat(currentSetpoint)
+	}
+}
+/*
+def modes() {
+	["home", "away", "vacation", "standby"]
+}*/
+
+
+def setThermostatMode() {
+	/*log.debug "switching thermostatMode"
+	def currentMode = device.currentState("thermostatMode")?.value
+	def modeOrder = modes()
+	def index = modeOrder.indexOf(currentMode)
+	def next = index >= 0 && index < modeOrder.size() - 1 ? modeOrder[index + 1] : modeOrder[0]
+	log.debug "switching mode from $currentMode to $next"
+	"$next"()*/
+}
+
+def setThermostatMode(def value) {
+	log.debug "setThermostatMode($value)"
+	/*def currentMode = device.currentState("thermostatMode")?.value
+	def lastTriedMode = state.lastTriedMode ?: currentMode ?: "heat"
+	def modeNumber;
+	Integer setpointModeNumber;
+	def modeToSendInString;*/
+	if (value == "heat") {
+		on()
+	}
+	else if (value == "cool") {
+		off()
+	}
+	else if (value == "on") {
+		on()
+	}
+	else if (value == "off") {
+		off()
+	}
+	else {
+		log.debug "MODE NOT SUPPORTED"
+	}
+	/*if (supportedModes?.contains(currentMode)) {
+		while (!supportedModes.contains(value) && value != "heat") {
+			value = next(value)
+		}
+	}
+	state.lastTriedMode = value
+	modeToSendInString = zigbee.convertToHexString(setpointModeNumber, 2)
+	delayBetween([
+			"st wattr 0x${device.deviceNetworkId} 0x19 0x201 0x001C 0x30 {$modeNumber}",
+			//my_writeAttribute(0x201, 0x401C, 0x30, modeToSendInString, ["mfgCode": "0x1185"]),
+			poll()
+	], 1000)*/
+}
+
+def on() {
+	log.debug "on"
+	sendEvent("name":"thermostatMode", "value":"heat")
+}
+
+def off() {
+	log.debug "off"
+	sendEvent("name":"thermostatMode", "value":"off")
+	//quickSetHeat(5)
+}
+
+def cool() {
+	log.debug "cool"
+	sendEvent("name":"thermostatMode", "value":"off")
+}
+
+def heat() {
+	log.debug "heat"
+	sendEvent("name":"thermostatMode", "value":"heat")
+}
+
+def emergencyHeat() {
+	log.debug "emergencyHeat"
+	sendEvent("name":"thermostatMode", "value":"emergency heat")
+	//"st wattr 0x${device.deviceNetworkId} 0x19 0x201 0x1C 0x30 {05}"
+}
+
+def setCustomThermostatMode(mode) {
+   setThermostatMode(mode)
+}
+/*
```

Comment:
If not needed the commented out code can be removed

Reply:
This was an handler offered to customer to be able to send specific modes in for special cases.
But other than that, it has no real use